### ISSUE TITLE
Retrospective gating with off-the-shelf components

### ISSUE THREAD

**Issue by Lewis-Kitchingman** [Lewis-Kitchingman/MIC-HACK-2025#3]
**Project Leader:** James Gholam (gholamja@cardiff.ac.uk)

**Project description**
Retrospective cardiac gating seeks to identify when, within a heartbeat, a particular MRI acquisition collected data. This information is used to predict how motion appears within the brain during motion-sensitive diffusion MRI. Our commercial ultra-low-field MRI system does not presently encoding the cardiac phase into the raw data it acquires. Additionally, it does not produce a timing signal to synchronise measurements with those from a pulse oximeter. In this project, we will attempt to analyse acoustic signals from the MRI, alongside real-time pulse measurements to produce estimates of cardiac-phase-labelled raw data.

**Ideal Participant Characteristics:**
Familiar with waveform and digital signal analysis in the time domain
Aware of MRI readouts, and of the nature of cardiac pulsation in the brain
Competent with microcontrollers like Arduino and Raspberry Pi
Literate in Python, C or Julia
Able to label waveforms indicating characteristic points of cardiac phases, and waveforms showing acoustic signals from the scanner (training provided)

**Resources:**
Libraries could include numpy, scipy, pytorch/tensorflow/flux etc for waveform analysis.
Pulse oximeter data will be collected in real time, whereas audio waveforms will be pre-recorded at the scanner.
Hardware used will include an Arduino and / or a Raspberry pi to record audio and collect pulse oximeter data, and a PulseSensor low-cost oximeter

**Project Plan:**
Task one:
Goal: Employ the PulseSensor low-cost pulse oximeter and microcontroller to collect real-time pulse measurements. Simultaneously record audio from a microphone
Steps: 
1. Use the libraries present at [https://pulsesensor.com/](https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fpulsesensor.com%2F&data=05%7C02%7CKitchingmanL%40cardiff.ac.uk%7C350fb99efecd4ad83bd908dd956c1646%7Cbdb74b3095684856bdbf06759778fcbc%7C1%7C0%7C638831013641636377%7CUnknown%7CTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7C0%7C%7C%7C&sdata=aJfawOJyCKKIiRWWuvEVv6OrOGywfyTKBuXytx8CIAI%3D&reserved=0) to record data from the oximeter
2. Use basic sensors to record audio (possibly using an external device such as a laptop)
3. Use the provided libraries to label data with cardiac phase
4. Establish a way of synchronising the two data streams

Expertise:
Familiarity with C, Arduino or Microcontrollers

**Task two:**
Goal: Analyse audio waveforms recorded from the ULF-MRI system engaged in scanning, and brainstorm a way of accurately identifying the start and end of readout
1. Label data from waveforms with start and end periods (training provided)
2. Brainstorm a way of identifying these points in longer time-series data automatically. This may include ML approaches, traditional time series analysis, spectral analysis etc
3. Apply this method on test data to see how accurately these points can be determined automatically

Expertise:
Familiar with time series data. Able to identify waveforms based on plots. Alternatively, familiar with time series analysis and / ML

**Task Three:** 
Goal: Package the code into a single program that can examine simultaneously recorded audio and oximeter data
1. Code a simple interface using e.g. ArgParse which will take in the unlabelled data, and produce synchronised timepoints for the cardiac cycle and MRI readout
Expertise:
Simple coding in Python etc to produce an easy-to-use tool

Tasks 1 and 2 can be conducted in parallel!
